rest_framework_tracking/management/commands/clearapilogs.py

```diff
@@ -0,0 +1,37 @@
+from django.core.management.base import BaseCommand, CommandError
+from rest_framework_tracking.models import APIRequestLog
+from optparse import make_option
+import datetime
+from django.utils import timezone
+
+class Command(BaseCommand):
+    help = 'Removes all api logs OR keeps X number of days of api logs'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--days_num',
+            help='Keep X number of days of logs and delete the rest',
+            type=int,
+            choices=[x + 1 for x in range(7)],
+        )
+
+    def handle(self, *args, **options):
+        days_num = options['days_num']
+
+        if days_num:
+            today = timezone.now()
+            start_date = today-datetime.timedelta(days=days_num)
+            logs_to_delete = APIRequestLog.objects.filter(requested_at__lt=start_date)
+
+        else:
+            logs_to_delete = APIRequestLog.objects.all().delete()
```

Comment:
return type of delete is a tuple with count of items and items deleted eg: (5, {'weblog.Blog': 1, 'weblog.Entry': 2, 'weblog.Entry_authors': 2}) (see: https://docs.djangoproject.com/en/3.1/ref/models/querysets/#delete)
and not a query set, the line logs_to_delete.delete() would fail....

Reply:
You are totally right. It is fixed now :)